src/libraries/System.Private.CoreLib/src/System/Diagnostics/CodeAnalysis/DynamicallyAccessedMemberTypes.cs

```diff
@@ -92,6 +92,76 @@ enum DynamicallyAccessedMemberTypes
         /// </summary>
         Interfaces = 0x2000,
 
+        /// <summary>
+        /// Specifies all non-public constructors, including those inherited from base classes.
+        /// </summary>
+        NonPublicConstructorsWithInherited = NonPublicConstructors | 0x4000,
```

Comment:
It was discussed here: https://www.youtube.com/watch?v=uyXehmcAkHc&t=0h9m31s.

We want a bit for each of these because e.g. if I want a combination of non-public fields (including inherited) and public constructors (but not inherited), this shouldn't mean to inherit all constructors. So we need an extra bit for each "inherited" possibility.

We also set the existing bit just for convenience so that bit operations still work nicely (i.e. it would be sufficient to do `NonPublicConstructorsWithInherited = 0x4000`, but `NonPublicConstructorsWithInherited = NonPublicConstructors | 0x4000` makes working with the enums easier because there's still the subset relationship and we can bit test without special casing.

The original proposal had two bits less (just "inherit constructors" and "inherit nested types") but the API review extended it to the more granular "inherit public constructors" and "inherit private constructors". Solely because the problem with .All is that it's too broad and we could have the same problem with "inherit constructors" being too broad in the future.

Reply:
I've moved the explanation to the PR description so that it's at the top.